src/v/redpanda/application.cc

```diff
@@ -559,6 +559,24 @@ void application::check_environment() {
     storage::directories::initialize(
       config::node().data_directory().as_sstring())
       .get();
+
+    if (config::shard_local_cfg().storage_strict_data_init()) {
+        // Look for the special file that indicates a user intends
+        // for the found data directory to be the one we use.
+        auto strict_data_dir_file
+          = config::node().strict_data_dir_file_path().string();
+        auto file_exists = ss::file_exists(strict_data_dir_file).get();
+
+        if (!file_exists) {
+            vlog(
```

Comment:
I don't think we need to log an error here: the exception thrown will get logged as an error immediately afterward.  Rather than logging two similar messages, let's just have one.

Reply:
Good point, removing the log.